lib/src/arithmetic/floating_point/floating_point_adder_simple.dart

```diff
@@ -47,82 +55,239 @@ class FloatingPointAdderSimple<FpType extends FloatingPoint>
+                  smaller.mantissa.getRange(0, mantissaWidth - 1),
+                  Const(0, width: 2)
+                ].swizzle()))
+        .named('smallMantissa');
+
+    final extendedWidth = min(
+        1 +
+            (mantissaWidth + 1) +
+            (a.runtimeType == FloatingPointExplicitJBit ? 2 : 0),
+        pow(2, exponentWidth).toInt() - 2);
+
+    final largeFinalMantissa = largeMantissa;
+    final smallExtendedMantissa = [
+      smallMantissa,
+      Const(0, width: extendedWidth)
+    ].swizzle().named('smallExtendedMantissa');
+    final smallShiftedMantissa =
+        (smallExtendedMantissa >>> expDiff).named('smallShiftedMantissa');
+
+    // Compute sticky bits past extendedWidth: expDiff - extendedWidth
+    final eW = Const(extendedWidth, width: expDiff.width).named('eW');
+    final rem =
+        mux(expDiff.gt(eW), expDiff - eW, Const(0, width: expDiff.width))
+            .named('rem');
+
+    final stickyBits =
+        (smallMantissa << Const(smallMantissa.width, width: rem.width) - rem)
+            .named('stickyBits');
+    final stickyBitsOr = stickyBits.or();
+
+    final largeNarrowMantissa = largeFinalMantissa;
+    final smallNarrowMantissa = smallShiftedMantissa.getRange(extendedWidth);
+
+    final highBitsAdder = CarrySelectOnesComplementCompoundAdder(
+        largeNarrowMantissa, smallNarrowMantissa,
+        generateCarryOut: true,
+        generateCarryOutP1: true,
+        subtractIn: effectiveSubtraction,
+        widthGen: widthGen,
+        adderGen: adderGen);
+
+    final carry = highBitsAdder.carryOut!;
+
+    final hSum = highBitsAdder.sum.named('highBitsSum');
+    final hSumP1 = highBitsAdder.sumP1.named('highBitsSumP1');
+
+    final lowerBits =
+        smallShiftedMantissa.getRange(0, extendedWidth).named('lowerBits');
+
+    final trueSign = mux(carry, larger.sign, smaller.sign).named('trueSign');
+
+    final lowBitsIncrement = (effectiveSubtraction & carry & expDiff.neq(0))
+        .named('lowBitsIncrement');
+
+    final carryBits = ~lowerBits.or();
+
+    final highBitsLSB =
+        (carryBits & (~stickyBitsOr & lowBitsIncrement)).named('highBitsLSB');
+
+    // TODO(desmonddak): This can work on narrow if not explicit-jbit
+    // We could optimize by splitting the search across the pipestage for
+    // high and low bits (low only matter for explicit-jbit)
+    final limitSize = smallShiftedMantissa.width;
+    final predictor = LeadingZeroAnticipate(
+        Const(0),
+        [largeFinalMantissa, Const(0, width: extendedWidth)]
+            .swizzle()
+            .slice(limitSize - 1, 0),
+        effectiveSubtraction,
+        smallShiftedMantissa.slice(limitSize - 1, 0),
+        endAroundCarry: carry);
+
+    final lead1Prediction = predictor.leadingOne!.named('lead1Prediction');
+
+    final lead1PredictionValid =
+        predictor.validLeadOne!.named('lead1PredictionValid');
+
+    final trueSignFlopped = localFlop(trueSign).named('trueSignFlopped');
```

Comment:
is extra `.named` really necessary since the condFlop already names them `_flopped`?

Reply:
Odd.  I must have done this before fixing condFlop or was just naming happy.